python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -281,6 +343,188 @@ def launch_scheduled_runs(
+        schedule_time_str = schedule_time.strftime(default_date_format_string())
+        if latest_tick and latest_tick.timestamp == schedule_timestamp:
+            tick = latest_tick
+            if latest_tick.status == TickStatus.FAILURE:
+                logger.info(f"Retrying previously failed schedule execution at {schedule_time_str}")
+            else:
+                logger.info(
+                    f"Resuming previously interrupted schedule execution at {schedule_time_str}"
+                )
+        else:
+            tick = instance.create_tick(
+                TickData(
+                    instigator_origin_id=instigator_origin_id,
+                    instigator_name=schedule_name,
+                    instigator_type=InstigatorType.SCHEDULE,
+                    status=TickStatus.STARTED,
+                    timestamp=schedule_timestamp,
+                    selector_id=external_schedule.selector_id,
+                )
+            )
+
+            _check_for_debug_crash(schedule_debug_crash_flags, "TICK_CREATED")
+
+        with _ScheduleLaunchContext(
+            external_schedule, tick, instance, logger, tick_retention_settings
+        ) as tick_context:
+            try:
+                _check_for_debug_crash(schedule_debug_crash_flags, "TICK_HELD")
+
+                yield from _schedule_runs_at_time(
+                    workspace_process_context,
+                    logger,
+                    external_schedule,
+                    schedule_time,
+                    tick_context,
+                    schedule_debug_crash_flags,
+                )
+            except Exception as e:
+                if isinstance(e, DagsterUserCodeUnreachableError):
+                    try:
+                        raise DagsterSchedulerError(
+                            f"Unable to reach the user code server for schedule {schedule_name}. Schedule will resume execution once the server is available."
+                        ) from e
+                    except:
+                        error_data = serializable_error_info_from_exc_info(sys.exc_info())
+
+                        tick_context.update_state(
+                            TickStatus.FAILURE,
+                            error=error_data,
+                            # don't increment the failure count - retry forever until the server comes back up
+                            # or the schedule is turned off
+                            failure_count=tick_context.failure_count,
+                        )
+                        yield error_data
+
+                else:
+                    error_data = serializable_error_info_from_exc_info(sys.exc_info())
+                    tick_context.update_state(
+                        TickStatus.FAILURE,
+                        error=error_data,
+                        failure_count=tick_context.failure_count + 1,
+                    )
+                    yield error_data
+
+
 def launch_scheduled_runs_for_schedule(
```

Comment:
still keeping all the code you duped - ideally try to minimize thrash to preserve blame